### PR TITLE
Spring Boot JWT completed

### PR DESCRIPTION
**Pull request by OleksiiZaloznyi**

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -39,7 +43,16 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 .authenticated()
                 )
                 .httpBasic(withDefaults())
+                .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+                .addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java]: let's add these to permitted
`          .requestMatchers("/auth/**", "/swagger-ui/**", "/v3/api-docs/**")`

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/test/resources/application.properties [hunk 2]
```diff
@@ -5,3 +5,6 @@ spring.datasource.password=password
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 
 spring.h2.console.enabled=true
+
+jwt.expiration=900000
+jwt.secret=hellomates1234567890asdfghjklwertyuisdsdsdsdsd
```
Comment (okuzan) [inline src/test/resources/application.properties]:
```suggestion
jwt.secret=hellomates1234567890asdfghjklwertyuisdsdsdsdsd

```
always have one empty line at the EOF

**Review by okuzan — APPROVED** [review]

File: src/main/java/com/example/bookstore/dto/category/CategoryRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+//package com.example.bookstore.dto.category;
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/dto/category/CategoryRequestDto.java:1]: ?

File: src/main/java/com/example/bookstore/dto/category/CategoryResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//package com.example.bookstore.dto.category;
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/dto/category/CategoryResponseDto.java:1]: ?

File: src/main/java/com/example/bookstore/dto/book/BookDtoWithoutCategoryIds.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    private String title;
+    @NotNull
+    private String author;
+    @NotNull
+    private String isbn;
+    @NotNull
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/dto/book/BookDtoWithoutCategoryIds.java:22]: empty line

File: src/main/java/com/example/bookstore/controller/CategoryController.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category Management", description = "Endpoints for managing categories")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/categories")
+public class CategoryController {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/controller/CategoryController.java:27]: empty line

File: src/main/java/com/example/bookstore/mapper/CategoryMapper.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.example.bookstore.config.MapperConfig;
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.model.Category;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(config = MapperConfig.class)
+public interface CategoryMapper {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/mapper/CategoryMapper.java:11]: empty line

File: src/main/java/com/example/bookstore/service/CategoryService.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.example.bookstore.service;
+
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.dto.category.CategoryDto;
+import java.util.List;
+
+public interface CategoryService {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/service/CategoryService.java:8]: empty line

File: src/main/java/com/example/bookstore/service/CategoryServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import com.example.bookstore.repository.CategoryRepository;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/service/CategoryServiceImpl.java:19]: empty line

File: src/main/resources/db/changelog/changes/13-add-book-to-category.yaml [hunk 10]
```diff
@@ -0,0 +1,50 @@
+#databaseChangeLog:
```
Comment (kozhukhovsky) [inline src/main/resources/db/changelog/changes/13-add-book-to-category.yaml:1]: ?

Comment (OleksiiZaloznyi) [reply src/main/resources/db/changelog/changes/13-add-book-to-category.yaml:1]: forgot to delete this

File: src/main/java/com/example/bookstore/dto/category/CategoryResponseDto.java (hunk 4 above)
Comment (OleksiiZaloznyi) [reply src/main/java/com/example/bookstore/dto/category/CategoryResponseDto.java:1]: forgot to delete this

File: src/main/java/com/example/bookstore/dto/category/CategoryRequestDto.java (hunk 3 above)
Comment (OleksiiZaloznyi) [reply src/main/java/com/example/bookstore/dto/category/CategoryRequestDto.java:1]: forgot to delete this

**Pull request merged**